src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -334,17 +334,27 @@ protected virtual void Initialize()
                     // If there's a parsing error, set up the logger and write out the previous messages so that they're
                     // discoverable in Application Insights. There's no filter, but that's okay since we cannot parse host.json to
                     // determine how to build the filtler.
-                    ConfigureLoggerFactory(ScriptConfig, FunctionTraceWriterFactory, _settingsManager, () => FileLoggingEnabled);
```

Comment:
Looks like that'll be tough here. I forgot the LoggerFactory is actually owned by the host config -- this 'configure' call sets things up on it. So we don't have a good hook point to sit with our Lazy. It may be a good candidate for a future refactoring, though...